localstack/packages/terraform.py

```diff
@@ -0,0 +1,42 @@
+import os
+import platform
+from typing import List
+
+from localstack.packages import InstallTarget, Package, PackageInstaller
+from localstack.packages.core import ArchiveDownloadAndExtractInstaller
+from localstack.utils.files import chmod_r
+from localstack.utils.platform import get_arch
+
+TERRAFORM_VERSION = "1.1.3"
+TERRAFORM_URL_TEMPLATE = (
+    "https://releases.hashicorp.com/terraform/{version}/terraform_{version}_{os}_{arch}.zip"
+)
+
+
+class TerraformPackage(Package):
+    def __init__(self):
+        super().__init__("Terraform", "1.1.3")
+
+    def get_versions(self) -> List[str]:
+        return ["1.1.3"]
+
+    def _get_installer(self, version: str) -> PackageInstaller:
+        return TerraformPackageInstaller("terraform", version)
+
+
+class TerraformPackageInstaller(ArchiveDownloadAndExtractInstaller):
+    def _get_install_marker_path(self, install_dir: str) -> str:
+        return os.path.join(install_dir, "terraform")
+
+    def _get_download_url(self) -> str:
+        system = platform.system().lower()
+        arch = get_arch()
+        return TERRAFORM_URL_TEMPLATE.format(version=TERRAFORM_VERSION, os=system, arch=arch)
+
+    def _install(self, target: InstallTarget) -> None:
+        super()._install(target)
+        chmod_r(self.get_executable_path(), 0o777)
+
+
+terraform_package = TerraformPackage()
+TERRAFORM_BIN = terraform_package.get_installer().get_executable_path()
```

Comment:
This line needs to be removed. The binary location needs to be evaluated at runtime when the binary is needed (or at least after it is ensured that it has been explicitly installed before), because with our new fallback mechanism the installation location is determined at the time of the installation.

Now, when terraform is not already installed at the time of the evaluation (i.e. when the file is imported the first time), this constant will have the value `None` and will stay `None`.

Reply:
:+1: missed that one. It wasn't even used, was just a remnant from trying to fix the test_terraform.py tests